File: android_mobile/src/test/java/com/alexstyl/specialdates/TestDateLabelCreator.java
package com.alexstyl.specialdates;

import com.alexstyl.specialdates.date.Date;
import com.alexstyl.specialdates.search.DateLabelCreator;
import com.alexstyl.specialdates.upcoming.MonthLabels;

import java.util.Locale;

final public class TestDateLabelCreator implements DateLabelCreator {

    private final MonthLabels monthLabels;

    public static DateLabelCreator forUS() {
        MonthLabels monthLabels = MonthLabels.forLocale(Locale.US);
        return new TestDateLabelCreator(monthLabels);
    }

    private TestDateLabelCreator(MonthLabels monthLabels) {
        this.monthLabels = monthLabels;
    }

    @Override
    public String createLabelWithoutYearFor(Date date) {
        return monthLabels.getMonthOfYear(date.getMonth()) + " " + date.getDayOfMonth();
    }

    @Override
    public String createLabelWithYearFor(Date date) {
        return monthLabels.getMonthOfYear(date.getMonth()) + " " + date.getDayOfMonth() + " " + date.getYear();
    }
}

File: android_mobile/src/test/java/com/alexstyl/specialdates/TestDateLabelCreator.kt
package com.alexstyl.specialdates

import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.date.DateLabelCreator
import com.alexstyl.specialdates.upcoming.MonthLabels
import java.util.Locale

class TestDateLabelCreator private constructor(private val monthLabels: MonthLabels) : DateLabelCreator {

    override fun createWithYearPreferred(date: Date): String {
        return if (date.hasYear()) {
            monthLabels.getMonthOfYear(date.month) + " " + date.dayOfMonth + " " + date.year
        } else {
            monthLabels.getMonthOfYear(date.month) + " " + date.dayOfMonth
        }
    }

    override fun createLabelWithoutYear(date: Date): String = monthLabels.getMonthOfYear(date.month) + " " + date.dayOfMonth

    companion object {
        fun forUS(): DateLabelCreator {
            val monthLabels = MonthLabels.forLocale(Locale.US)
            return TestDateLabelCreator(monthLabels)
        }
    }
}
